client.go

```diff
@@ -336,41 +363,48 @@ func (c *client) LocateRegion(table, row []byte, useCache bool) *RegionInfo {
 		},
 	})
 
-	conn.call(call)
-	// TODO err handling
+	err = conn.call(call)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
 	response := <-call.responseCh
 
 	switch r := response.(type) {
 	case *proto.GetResponse:
 		res := r.GetResult()
 		if res == nil {
-			log.Warnf("Couldn't find the region: [table=%s] [row=%s] [region_row=%s]", table, row, regionRow)
-			return nil
+			return nil, errors.Errorf("Empty region: [table=%s] [row=%q] [region_row=%q]", table, row, regionRow)
 		}
+
 		rr := NewResultRow(res)
-		if region := c.parseRegion(rr); region != nil {
-			c.updateRegionCache(table, region)
-			return region
+		region, err := c.parseRegion(rr)
+		if err != nil {
+			return nil, errors.Trace(err)
 		}
+
+		c.updateRegionCache(table, region)
+		return region, nil
+	case *exception:
+		return nil, errors.New(r.msg)
+	default:
+		log.Warnf("Unknown response - %T - %v", r, r)
 	}
-	return nil
+
+	return nil, errors.Errorf("Couldn't find the region: [table=%s] [row=%q] [region_row=%q]", table, row, regionRow)
 }
 
 func (c *client) Close() error {
 	if c.zkClient != nil {
 		c.zkClient.Close()
 	}
-	var err error
-	if c.cachedConns != nil && len(c.cachedConns) > 0 {
-		for _, conn := range c.cachedConns {
-			err = conn.close()
-			if err != nil {
-				err = errors.Trace(err)
-			}
+
+	for _, conn := range c.cachedConns {
+		err := conn.close()
+		if err != nil {
+			return errors.Trace(err)
```

Comment:
Yes, so u should be careful if u call Close function, but most of time it is OK i think,  Close function usually be called before exit.